*** inflate.c.orig	2019-11-16 15:17:06.548780288 +0100
--- inflate.c	2019-11-16 15:17:06.562780246 +0100
***************
*** 30,36 ****
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
- #include <stdlib.h>
  #include "matio_private.h"
  
  #if HAVE_ZLIB
--- 30,35 ----
***************
*** 381,390 ****
          Mat_int32Swap(tag);
          Mat_int32Swap(tag+1);
      }
!     if ( (tag[0] & 0x0000ffff) != MAT_T_INT32 ) {
!         Mat_Critical("InflateDimensions: Reading dimensions expected type MAT_T_INT32");
          return bytesread;
-     }
      rank = tag[1];
      if ( rank % 8 != 0 )
          i = 8-(rank %8);
--- 380,392 ----
          Mat_int32Swap(tag);
          Mat_int32Swap(tag+1);
      }
! 
!     /**
!      *  Stefan Widgren: 2014-04-28
!      *  Removed throwing an error if the data type not equals MAT_T_INT32
!      */
!     if ( (tag[0] & 0x0000ffff) != MAT_T_INT32 )
          return bytesread;
      rank = tag[1];
      if ( rank % 8 != 0 )
          i = 8-(rank %8);
