*** inflate.c.orig	2017-07-21 11:11:55.858427517 +0200
--- inflate.c	2017-07-21 11:11:55.890427846 +0200
***************
*** 30,36 ****
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
! #include <stdlib.h>
  #include "matio_private.h"
  
  #if HAVE_ZLIB
--- 30,46 ----
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
! /*
!  * Changes in the R package rmatio:
!  *
!  * - The io routines have been adopted to use R printing and error routines.
!  *   See the R manual Writing R Extensions
!  *
!  */
! 
! #include <Rdefines.h>
! #define Mat_Critical error
! 
  #include "matio_private.h"
  
  #if HAVE_ZLIB
***************
*** 380,389 ****
          Mat_int32Swap(tag);
          Mat_int32Swap(tag+1);
      }
!     if ( (tag[0] & 0x0000ffff) != MAT_T_INT32 ) {
!         Mat_Critical("InflateDimensions: Reading dimensions expected type MAT_T_INT32");
          return bytesread;
-     }
      rank = tag[1];
      if ( rank % 8 != 0 )
          i = 8-(rank %8);
--- 390,402 ----
          Mat_int32Swap(tag);
          Mat_int32Swap(tag+1);
      }
! 
!     /**
!      *  Stefan Widgren: 2014-04-28
!      *  Removed throwing an error if the data type not equals MAT_T_INT32
!      */
!     if ( (tag[0] & 0x0000ffff) != MAT_T_INT32 )
          return bytesread;
      rank = tag[1];
      if ( rank % 8 != 0 )
          i = 8-(rank %8);
