*** inflate.c.orig	2017-07-27 18:13:16.006895407 +0200
--- inflate.c	2017-07-27 18:12:29.806948269 +0200
***************
*** 30,36 ****
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
- #include <stdlib.h>
  #include "matio_private.h"
  
  #if HAVE_ZLIB
--- 30,35 ----
***************
*** 380,389 ****
          Mat_int32Swap(tag);
          Mat_int32Swap(tag+1);
      }
!     if ( (tag[0] & 0x0000ffff) != MAT_T_INT32 ) {
!         Mat_Critical("InflateDimensions: Reading dimensions expected type MAT_T_INT32");
          return bytesread;
-     }
      rank = tag[1];
      if ( rank % 8 != 0 )
          i = 8-(rank %8);
--- 379,391 ----
          Mat_int32Swap(tag);
          Mat_int32Swap(tag+1);
      }
! 
!     /**
!      *  Stefan Widgren: 2014-04-28
!      *  Removed throwing an error if the data type not equals MAT_T_INT32
!      */
!     if ( (tag[0] & 0x0000ffff) != MAT_T_INT32 )
          return bytesread;
      rank = tag[1];
      if ( rank % 8 != 0 )
          i = 8-(rank %8);
