*** mat4.c.orig	2017-07-21 22:31:21.650945646 +0200
--- mat4.c	2017-07-21 22:22:18.981770122 +0200
***************
*** 30,35 ****
--- 30,46 ----
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
+ /*
+  * Changes in the R package rmatio:
+  *
+  * - The io routines have been adopted to use R printing and error routines.
+  *   See the R manual Writing R Extensions
+  *
+  */
+ 
+ #include <Rdefines.h>
+ #define Mat_Critical error
+ 
  #include <stdlib.h>
  #include <string.h>
  #include <stdio.h>
***************
*** 73,79 ****
      mat->bof           = 0;
      mat->next_index    = 0;
      mat->refs_id       = -1;
!     mat->filename      = strdup_printf("%s",matname);
      mat->mode          = 0;
  
      Mat_Rewind(mat);
--- 84,92 ----
      mat->bof           = 0;
      mat->next_index    = 0;
      mat->refs_id       = -1;
!     /* Stefan Widgren 2017-07-20 Replaced strdup_printf with strdup */
!     /* mat->filename = strdup_printf("%s",matname); */
!     mat->filename      = strdup(matname);
      mat->mode          = 0;
  
      Mat_Rewind(mat);
