*** mat5.h.orig	2017-07-20 12:51:36.162270629 +0200
--- mat5.h	2017-07-20 12:59:39.590026174 +0200
***************
*** 34,67 ****
  #   define EXTERN extern
  #endif
  
! static size_t GetStructFieldBufSize(matvar_t *matvar);
! static size_t GetCellArrayFieldBufSize(matvar_t *matvar);
! static size_t GetMatrixMaxBufSize(matvar_t *matvar);
! static size_t GetEmptyMatrixMaxBufSize(const char *name,int rank);
! static int WriteEmptyCharData(mat_t *mat, int N, enum matio_types data_type);
! static int WriteEmptyData(mat_t *mat,int N,enum matio_types data_type);
! static int ReadNextCell( mat_t *mat, matvar_t *matvar );
! static int ReadNextStructField( mat_t *mat, matvar_t *matvar );
! static int ReadNextFunctionHandle(mat_t *mat, matvar_t *matvar);
! static int WriteCellArrayFieldInfo(mat_t *mat,matvar_t *matvar);
! static int WriteCellArrayField(mat_t *mat,matvar_t *matvar );
! static int WriteStructField(mat_t *mat,matvar_t *matvar);
! static size_t Mat_WriteEmptyVariable5(mat_t *mat,const char *name,int rank,
!                   size_t *dims);
! #if defined(HAVE_ZLIB)
! static size_t WriteCompressedCharData(mat_t *mat,z_stream *z,void *data,int N,
!                   enum matio_types data_type);
! static int    WriteCompressedEmptyData(mat_t *mat,z_stream *z,int N,
!                   enum matio_types data_type);
! static size_t WriteCompressedData(mat_t *mat,z_stream *z,void *data,int N,
!                   enum matio_types data_type);
! static size_t WriteCompressedCellArrayField(mat_t *mat,matvar_t *matvar,
!                   z_stream *z);
! static size_t WriteCompressedStructField(mat_t *mat,matvar_t *matvar,
!                   z_stream *z);
! static size_t Mat_WriteCompressedEmptyVariable5(mat_t *mat,const char *name,
!                   int rank,size_t *dims,z_stream *z);
! #endif
  
  /*   mat5.c    */
  EXTERN mat_t *Mat_Create5(const char *matname,const char *hdr_str);
--- 34,69 ----
  #   define EXTERN extern
  #endif
  
! /* Stefan Widgren 2014-01-05: Moved the following forward declarations
!  * to mat5.c to silent compiler warnings on windows*/
! /* static size_t GetStructFieldBufSize(matvar_t *matvar); */
! /* static size_t GetCellArrayFieldBufSize(matvar_t *matvar); */
! /* static size_t GetMatrixMaxBufSize(matvar_t *matvar); */
! /* static size_t GetEmptyMatrixMaxBufSize(const char *name,int rank); */
! /* static int WriteEmptyCharData(mat_t *mat, int N, enum matio_types data_type); */
! /* static int WriteEmptyData(mat_t *mat,int N,enum matio_types data_type); */
! /* static int ReadNextCell( mat_t *mat, matvar_t *matvar ); */
! /* static int ReadNextStructField( mat_t *mat, matvar_t *matvar ); */
! /* static int ReadNextFunctionHandle(mat_t *mat, matvar_t *matvar); */
! /* static int WriteCellArrayFieldInfo(mat_t *mat,matvar_t *matvar); */
! /* static int WriteCellArrayField(mat_t *mat,matvar_t *matvar ); */
! /* static int WriteStructField(mat_t *mat,matvar_t *matvar); */
! /* static size_t Mat_WriteEmptyVariable5(mat_t *mat,const char *name,int rank, */
! /*                   size_t *dims); */
! /* #if defined(HAVE_ZLIB) */
! /* static size_t WriteCompressedCharData(mat_t *mat,z_stream *z,void *data,int N, */
! /*                   enum matio_types data_type); */
! /* static int    WriteCompressedEmptyData(mat_t *mat,z_stream *z,int N, */
! /*                   enum matio_types data_type); */
! /* static size_t WriteCompressedData(mat_t *mat,z_stream *z,void *data,int N, */
! /*                   enum matio_types data_type); */
! /* static size_t WriteCompressedCellArrayField(mat_t *mat,matvar_t *matvar, */
! /*                   z_stream *z); */
! /* static size_t WriteCompressedStructField(mat_t *mat,matvar_t *matvar, */
! /*                   z_stream *z); */
! /* static size_t Mat_WriteCompressedEmptyVariable5(mat_t *mat,const char *name, */
! /*                   int rank,size_t *dims,z_stream *z); */
! /* #endif */
  
  /*   mat5.c    */
  EXTERN mat_t *Mat_Create5(const char *matname,const char *hdr_str);
