*** mat5.h.orig	2017-07-21 09:25:23.975395321 +0200
--- mat5.h	2017-07-21 09:31:05.251530433 +0200
***************
*** 34,68 ****
  #   define EXTERN extern
  #endif
  
! static size_t GetStructFieldBufSize(matvar_t *matvar);
! static size_t GetCellArrayFieldBufSize(matvar_t *matvar);
! static size_t GetMatrixMaxBufSize(matvar_t *matvar);
! static size_t GetEmptyMatrixMaxBufSize(const char *name,int rank);
! static size_t WriteCharData(mat_t *mat, void *data, int N,enum matio_types data_type);
! static size_t WriteEmptyCharData(mat_t *mat, int N, enum matio_types data_type);
! static size_t WriteEmptyData(mat_t *mat,int N,enum matio_types data_type);
! static size_t ReadNextCell( mat_t *mat, matvar_t *matvar );
! static size_t ReadNextStructField( mat_t *mat, matvar_t *matvar );
! static size_t ReadNextFunctionHandle(mat_t *mat, matvar_t *matvar);
! static int WriteCellArrayFieldInfo(mat_t *mat,matvar_t *matvar);
! static int WriteCellArrayField(mat_t *mat,matvar_t *matvar );
! static int WriteStructField(mat_t *mat,matvar_t *matvar);
! static size_t Mat_WriteEmptyVariable5(mat_t *mat,const char *name,int rank,
!                   size_t *dims);
! #if defined(HAVE_ZLIB)
! static size_t WriteCompressedCharData(mat_t *mat,z_streamp z,void *data,int N,
!                   enum matio_types data_type);
! static size_t WriteCompressedEmptyData(mat_t *mat,z_streamp z,int N,
!                   enum matio_types data_type);
! static size_t WriteCompressedData(mat_t *mat,z_streamp z,void *data,int N,
!                   enum matio_types data_type);
! static size_t WriteCompressedCellArrayField(mat_t *mat,matvar_t *matvar,
!                   z_streamp z);
! static size_t WriteCompressedStructField(mat_t *mat,matvar_t *matvar,
!                   z_streamp z);
! static size_t Mat_WriteCompressedEmptyVariable5(mat_t *mat,const char *name,
!                   int rank,size_t *dims,z_streamp z);
! #endif
  
  /*   mat5.c    */
  EXTERN mat_t *Mat_Create5(const char *matname,const char *hdr_str);
--- 34,70 ----
  #   define EXTERN extern
  #endif
  
! /* Stefan Widgren 2014-01-05: Moved the following forward declarations
!  * to mat5.c to silent compiler warnings on windows*/
! /* static size_t GetStructFieldBufSize(matvar_t *matvar); */
! /* static size_t GetCellArrayFieldBufSize(matvar_t *matvar); */
! /* static size_t GetMatrixMaxBufSize(matvar_t *matvar); */
! /* static size_t GetEmptyMatrixMaxBufSize(const char *name,int rank); */
! /* static size_t WriteCharData(mat_t *mat, void *data, int N,enum matio_types data_type); */
! /* static size_t WriteEmptyCharData(mat_t *mat, int N, enum matio_types data_type); */
! /* static size_t WriteEmptyData(mat_t *mat,int N,enum matio_types data_type); */
! /* static size_t ReadNextCell( mat_t *mat, matvar_t *matvar ); */
! /* static size_t ReadNextStructField( mat_t *mat, matvar_t *matvar ); */
! /* static size_t ReadNextFunctionHandle(mat_t *mat, matvar_t *matvar); */
! /* static int WriteCellArrayFieldInfo(mat_t *mat,matvar_t *matvar); */
! /* static int WriteCellArrayField(mat_t *mat,matvar_t *matvar ); */
! /* static int WriteStructField(mat_t *mat,matvar_t *matvar); */
! /* static size_t Mat_WriteEmptyVariable5(mat_t *mat,const char *name,int rank, */
! /*                   size_t *dims); */
! /* #if defined(HAVE_ZLIB) */
! /* static size_t WriteCompressedCharData(mat_t *mat,z_streamp z,void *data,int N, */
! /*                   enum matio_types data_type); */
! /* static size_t WriteCompressedEmptyData(mat_t *mat,z_streamp z,int N, */
! /*                   enum matio_types data_type); */
! /* static size_t WriteCompressedData(mat_t *mat,z_streamp z,void *data,int N, */
! /*                   enum matio_types data_type); */
! /* static size_t WriteCompressedCellArrayField(mat_t *mat,matvar_t *matvar, */
! /*                   z_streamp z); */
! /* static size_t WriteCompressedStructField(mat_t *mat,matvar_t *matvar, */
! /*                   z_streamp z); */
! /* static size_t Mat_WriteCompressedEmptyVariable5(mat_t *mat,const char *name, */
! /*                   int rank,size_t *dims,z_streamp z); */
! /* #endif */
  
  /*   mat5.c    */
  EXTERN mat_t *Mat_Create5(const char *matname,const char *hdr_str);
