*** matio.h.orig	2017-07-21 10:01:54.498528925 +0200
--- matio.h	2017-07-21 10:01:30.349576523 +0200
***************
*** 30,43 ****
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
  #ifndef MATIO_H
  #define MATIO_H
  
  #include <stdlib.h>
  #include <stdio.h>
! #include "matio_pubconf.h"
! 
! #include <stdarg.h>
  
  #ifdef __cplusplus
  #   define EXTERN extern "C"
--- 30,48 ----
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
+ /*
+  * Changes in the R package rmatio:
+  *
+  * - The io routines have been adopted to use R printing and error routines.
+  *   See the R manual Writing R Extensions
+  */
+ 
  #ifndef MATIO_H
  #define MATIO_H
  
  #include <stdlib.h>
  #include <stdio.h>
! #include "config.h"
  
  #ifdef __cplusplus
  #   define EXTERN extern "C"
***************
*** 123,129 ****
      MAT_C_UINT32   = 13, /**< @brief Matlab unsigned 32-bit integer class  */
      MAT_C_INT64    = 14, /**< @brief Matlab signed 64-bit integer class    */
      MAT_C_UINT64   = 15, /**< @brief Matlab unsigned 64-bit integer class  */
!     MAT_C_FUNCTION = 16  /**< @brief Matlab function class                 */
  };
  
  /** @brief Matlab array flags
--- 128,135 ----
      MAT_C_UINT32   = 13, /**< @brief Matlab unsigned 32-bit integer class  */
      MAT_C_INT64    = 14, /**< @brief Matlab signed 64-bit integer class    */
      MAT_C_UINT64   = 15, /**< @brief Matlab unsigned 64-bit integer class  */
!     MAT_C_FUNCTION = 16, /**< @brief Matlab function class                 */
!     MAT_C_OPAQUE   = 17  /**< @brief Matlab function workspace?            */
  };
  
  /** @brief Matlab array flags
***************
*** 223,244 ****
  /* Library function */
  EXTERN void Mat_GetLibraryVersion(int *major,int *minor,int *release);
  
  /*     io.c         */
! EXTERN char  *strdup_vprintf(const char *format, va_list ap);
! EXTERN char  *strdup_printf(const char *format, ...);
! EXTERN int    Mat_SetVerbose( int verb, int s );
! EXTERN int    Mat_SetDebug( int d );
! EXTERN void   Mat_Critical( const char *format, ... );
! EXTERN MATIO_NORETURN void Mat_Error( const char *format, ... ) MATIO_NORETURNATTR;
! EXTERN void   Mat_Help( const char *helpstr[] );
! EXTERN int    Mat_LogInit( const char *progname );
! EXTERN int    Mat_LogClose(void);
! EXTERN int    Mat_LogInitFunc(const char *prog_name,
!                     void (*log_func)(int log_level, char *message) );
! EXTERN int    Mat_Message( const char *format, ... );
! EXTERN int    Mat_DebugMessage( int level, const char *format, ... );
! EXTERN int    Mat_VerbMessage( int level, const char *format, ... );
! EXTERN void   Mat_Warning( const char *format, ... );
  EXTERN size_t Mat_SizeOf(enum matio_types data_type);
  EXTERN size_t Mat_SizeOfClass(int class_type);
  
--- 229,254 ----
  /* Library function */
  EXTERN void Mat_GetLibraryVersion(int *major,int *minor,int *release);
  
+ /* Stefan Widgren 2014-01-01: Not used by rmatio */
  /*     io.c         */
! /* EXTERN char  *strdup_vprintf(const char *format, va_list ap); */
! /* EXTERN char  *strdup_printf(const char *format, ...); */
! /* EXTERN int    Mat_SetVerbose( int verb, int s ); */
! /* EXTERN int    Mat_SetDebug( int d ); */
! /* EXTERN void   Mat_Critical( const char *format, ... ); */
! /* EXTERN void   Mat_Error( const char *format, ... ); */
! /* EXTERN void   Mat_Help( const char *helpstr[] ); */
! /* EXTERN int    Mat_LogInit( const char *progname ); */
! /* EXTERN int    Mat_LogClose(void); */
! /* EXTERN int    Mat_LogInitFunc(const char *prog_name, */
! /*                     void (*log_func)(int log_level, char *message) ); */
! /* EXTERN int    Mat_Message( const char *format, ... ); */
! /* EXTERN int    Mat_DebugMessage( int level, const char *format, ... ); */
! /* EXTERN int    Mat_VerbMessage( int level, const char *format, ... ); */
! /* EXTERN void   Mat_Warning( const char *format, ... ); */
! /* EXTERN void   Mat_PrintNumber(enum matio_types type, void *data); */
! /* EXTERN void   Mat_VarPrint( matvar_t *matvar, int printdata ); */
! 
  EXTERN size_t Mat_SizeOf(enum matio_types data_type);
  EXTERN size_t Mat_SizeOfClass(int class_type);
  
***************
*** 260,266 ****
                        void *data,int opt);
  EXTERN matvar_t  *Mat_VarCreateStruct(const char *name,int rank,size_t *dims,
                        const char **fields,unsigned nfields);
! EXTERN int        Mat_VarDelete(mat_t *mat, const char *name);
  EXTERN matvar_t  *Mat_VarDuplicate(const matvar_t *in, int opt);
  EXTERN void       Mat_VarFree(matvar_t *matvar);
  EXTERN matvar_t  *Mat_VarGetCell(matvar_t *matvar,int index);
--- 270,281 ----
                        void *data,int opt);
  EXTERN matvar_t  *Mat_VarCreateStruct(const char *name,int rank,size_t *dims,
                        const char **fields,unsigned nfields);
! 
! /* Stefan Widgren 2014-01-05: Commented out the following unused
!  * function in the rmatio package to silent compiler warning (implicit
!  * declaration of mktemp) on windows. */
! /* EXTERN int        Mat_VarDelete(mat_t *mat, const char *name); */
! 
  EXTERN matvar_t  *Mat_VarDuplicate(const matvar_t *in, int opt);
  EXTERN void       Mat_VarFree(matvar_t *matvar);
  EXTERN matvar_t  *Mat_VarGetCell(matvar_t *matvar,int index);
***************
*** 282,288 ****
                        int *edge,int copy_fields);
  EXTERN matvar_t  *Mat_VarGetStructsLinear(matvar_t *matvar,int start,int stride,
                        int edge,int copy_fields);
- EXTERN void       Mat_VarPrint( matvar_t *matvar, int printdata );
  EXTERN matvar_t  *Mat_VarRead(mat_t *mat, const char *name );
  EXTERN int        Mat_VarReadData(mat_t *mat,matvar_t *matvar,void *data,
                        int *start,int *stride,int *edge);
--- 297,302 ----
