*** matvar_struct.c.orig	2017-07-20 12:51:36.162270629 +0200
--- matvar_struct.c	2017-07-20 13:00:28.744279874 +0200
***************
*** 25,30 ****
--- 25,42 ----
   * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
   * POSSIBILITY OF SUCH DAMAGE.
   */
+ 
+ /*
+  * Changes in the R package rmatio:
+  *
+  * - The io routines have been adopted to use R printing and error routines.
+  *   See the R manual Writing R Extensions
+  *
+  */
+ 
+ #include <Rdefines.h>
+ #define Mat_Critical error
+ 
  #include <stdlib.h>
  #include <string.h>
  #include "matio_private.h"
