*** matvar_struct.c.orig	2017-07-21 09:25:23.967391864 +0200
--- matvar_struct.c	2017-07-21 09:35:44.518314397 +0200
***************
*** 26,31 ****
--- 26,42 ----
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
+ /*
+  * Changes in the R package rmatio:
+  *
+  * - The io routines have been adopted to use R printing and error routines.
+  *   See the R manual Writing R Extensions
+  *
+  */
+ 
+ #include <Rdefines.h>
+ #define Mat_Critical error
+ 
  #include <stdlib.h>
  #include <string.h>
  #include "matio_private.h"
