*** read_data.c.orig	2019-11-16 15:49:23.921611107 +0100
--- read_data.c	2019-11-16 15:49:23.927611091 +0100
***************
*** 30,45 ****
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
! /* FIXME: Implement Unicode support */
! #include <stdlib.h>
! #include <string.h>
! #include <stdio.h>
! #include <math.h>
! #include <time.h>
  #include "matio_private.h"
! #if defined(HAVE_ZLIB)
! #   include <zlib.h>
! #endif
  
  #if !defined(READ_BLOCK_SIZE)
  #define READ_BLOCK_SIZE (256)
--- 30,58 ----
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
! /*
!  * Changes in the R package rmatio:
!  *
!  * - The io routines have been adopted to use R printing and error routines.
!  *   See the R manual Writing R Extensions
!  *
!  */
! 
! /* Stefan Widgren 2014-01-04: Include files for rmatio package */
! #include <R.h>
  #include "matio_private.h"
! 
! /* FIXME: Implement Unicode support */
! /* #include <stdlib.h> */
! /* #include <string.h> */
! /* #include <stdio.h> */
! /* #include <math.h> */
! /* #include <time.h> */
! /* #include "matio_private.h" */
! 
! /* #if defined(HAVE_ZLIB) */
! /* #   include <zlib.h> */
! /* #endif */
  
  #if !defined(READ_BLOCK_SIZE)
  #define READ_BLOCK_SIZE (256)
***************
*** 2063,2076 ****
      enum matio_classes class_type,enum matio_types data_type,int start,
      int stride,int edge)
  {
!     int nBytes = 0, i, err;
      z_stream z_copy = {0,};
  
      if ( (mat   == NULL) || (data   == NULL) || (mat->fp == NULL) )
          return 0;
  
      stride--;
!     err = inflateCopy(&z_copy,z);
      InflateSkipData(mat,&z_copy,data_type,start);
      switch ( class_type ) {
          case MAT_C_DOUBLE:
--- 2076,2089 ----
      enum matio_classes class_type,enum matio_types data_type,int start,
      int stride,int edge)
  {
!     int nBytes = 0, i;
      z_stream z_copy = {0,};
  
      if ( (mat   == NULL) || (data   == NULL) || (mat->fp == NULL) )
          return 0;
  
      stride--;
!     inflateCopy(&z_copy,z);
      InflateSkipData(mat,&z_copy,data_type,start);
      switch ( class_type ) {
          case MAT_C_DOUBLE:
***************
*** 2200,2206 ****
      enum matio_classes class_type,enum matio_types data_type,size_t *dims,
      int *start,int *stride,int *edge)
  {
!     int nBytes = 0, i, j, err;
      int pos, row_stride, col_stride;
      z_stream z_copy = {0,};
  
--- 2213,2219 ----
      enum matio_classes class_type,enum matio_types data_type,size_t *dims,
      int *start,int *stride,int *edge)
  {
!     int nBytes = 0, i, j;
      int pos, row_stride, col_stride;
      z_stream z_copy = {0,};
  
***************
*** 2209,2215 ****
          return 0;
      }
  
!     err = inflateCopy(&z_copy,z);
      switch ( class_type ) {
          case MAT_C_DOUBLE:
          {
--- 2222,2228 ----
          return 0;
      }
  
!     inflateCopy(&z_copy,z);
      switch ( class_type ) {
          case MAT_C_DOUBLE:
          {
