*** read_data.c.orig	2017-07-28 16:00:19.310938201 +0200
--- read_data.c	2017-07-28 16:00:19.322938291 +0200
***************
*** 30,45 ****
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
! /* FIXME: Implement Unicode support */
! #include <stdlib.h>
! #include <string.h>
! #include <stdio.h>
! #include <math.h>
! #include <time.h>
  #include "matio_private.h"
! #if defined(HAVE_ZLIB)
! #   include <zlib.h>
! #endif
  
  #define READ_DATA(SwapFunc) \
      do { \
--- 30,58 ----
   * POSSIBILITY OF SUCH DAMAGE.
   */
  
! /*
!  * Changes in the R package rmatio:
!  *
!  * - The io routines have been adopted to use R printing and error routines.
!  *   See the R manual Writing R Extensions
!  *
!  */
! 
! /* Stefan Widgren 2014-01-04: Include files for rmatio package */
! #include <R.h>
  #include "matio_private.h"
! 
! /* FIXME: Implement Unicode support */
! /* #include <stdlib.h> */
! /* #include <string.h> */
! /* #include <stdio.h> */
! /* #include <math.h> */
! /* #include <time.h> */
! /* #include "matio_private.h" */
! 
! /* #if defined(HAVE_ZLIB) */
! /* #   include <zlib.h> */
! /* #endif */
  
  #define READ_DATA(SwapFunc) \
      do { \
